cli/cmd/plugin/ui/web/tanzu-ui/src/views/unmanaged-cluster/UnmanagedClusterInventory.tsx

```diff
@@ -16,62 +17,193 @@ import './UnmanagedClusterInventory.scss';
 
 function UnmanagedClusterInventory() {
     const [unmanagedClusters, setUnmanagedClusters] = useState<UnmanagedCluster[]>([]);
+    const [showDeleteModal, setShowDeleteModal] = useState<boolean>(false);
+    const [clusterNameForDeletion, setClusterNameForDeletion] = useState<string>('');
     const navigate = useNavigate();
 
+    function retrieveUnmanagedClusters() {
+        UnmanagedService.getUnmanagedClusters().then(
+            (data) => {
+                setUnmanagedClusters(data);
+            },
+            (err) => {
+                console.warn(`unmanaged clusters get api failed with error: ${err}`);
+            }
+        );
+    }
+
     useEffect(() => {
-        // fetch unmanaged clusters
-        UnmanagedService.getUnmanagedClusters().then((data) => setUnmanagedClusters(data));
+        retrieveUnmanagedClusters();
     }, []);
 
+    function showConfirmDeleteModal(clusterName?: string) {
+        if (!clusterName) {
+            console.warn(`toggleConfirmDeleteModal expected a cluster name string; instead got ${clusterName}`);
+            return;
+        }
+        setClusterNameForDeletion(clusterName);
+        setShowDeleteModal(!showDeleteModal);
+    }
+
+    function hasUnmanagedClusters() {
+        return unmanagedClusters.length ? true : false;
+    }
+
+    function deleteUnmanagedCluster(clusterName: string) {
+        setShowDeleteModal(!showDeleteModal);
```

Comment:
I would consider just `setShowDeleteModal(false)`, since that should be the same (unless I'm missing something), and it's easier to understand. Also, separate UX question (for another PR), would it make sense to have a spinner show WHILE the cluster is being deleted? Or does it happen too quickly to care about that?